cmake/deps/json.cmake

```
@@ -1,10 +1,14 @@
 include(FetchContent)
 
-set(JSON_BuildTests OFF)
-set(JSON_Install OFF)
+find_package(nlohmann_json CONFIG QUIET)
+if (NOT nlohmann_json_FOUND)
+  set(JSON_BuildTests OFF)
+  set(JSON_Install OFF)
 
-FetchContent_Declare(json
-  URL https://github.com/nlohmann/json/releases/download/v3.11.2/json.tar.xz
-)
+  FetchContent_Declare(nlohmann_json
+    URL https://github.com/nlohmann/json/archive/refs/tags/v3.12.0.tar.gz
+    URL_HASH SHA256=4b92eb0c06d10683f7447ce9406cb97cd4b453be18d7279320f7b2f025c10187
+  )
```

Alright, I did [my homework](https://cmake.org/cmake/help/latest/guide/using-dependencies/index.html#fetchcontent-and-find-package-integration). I guess a simpler solution (and more idiomatic) solution is:

```suggestion
  FetchContent_Declare(nlohmann_json
    URL https://github.com/nlohmann/json/archive/refs/tags/v3.12.0.tar.gz
    URL_HASH SHA256=4b92eb0c06d10683f7447ce9406cb97cd4b453be18d7279320f7b2f025c10187
    FIND_PACKAGE_ARGS NAMES nlohmann_json
  )
```

Adding `FIND_PACKAGE_ARGS` will help `FetchContent_*` to satisfy the dependency o with `find_package()`. If that dependency is not found, then it is built from source.

Can you have a try?

---

Yep, I see that you require cmake >= 3.24, so `FIND_PACKAGE_ARGS` will be available.